### PR TITLE
RUST-1401 Additional GridFS API

### PR DESCRIPTION
**Pull request by isabelatkinson**
Implements the remaining API methods defined in the GridFS spec, including the advanced API.

### CODE REVIEW

File: src/gridfs.rs [hunk 1]
```diff
@@ -318,28 +329,58 @@ impl GridFsBucket {
...
-    /// Given an `id`, deletes the stored file's files collection document and
-    /// associated chunks from a [`GridFsBucket`].
-    pub async fn delete(&self, id: Bson) {
-        todo!()
+    /// Deletes the [`FilesCollectionDocument`] with the given `id `and its associated chunks from
+    /// this bucket.
+    pub async fn delete(&self, id: Bson) -> Result<()> {
+        let delete_result = self
+            .files()
+            .delete_one(doc! { "_id": id.clone() }, None)
```
Comment (abr-egn) [inline src/gridfs.rs:337]: Just to validate - `delete` doesn't return an error if zero documents match, correct?

**Review by abr-egn — APPROVED** [review]
LGTM!

**Review by kmahar — COMMENTED** [review]
one comment but LGTM otherwise

**Review by patrickfreed — APPROVED** [review]

File: src/gridfs/options.rs [hunk 2]
```diff
@@ -82,9 +82,28 @@ pub struct GridFsFindOptions {
     /// The maximum amount of time to allow the query to run.
     pub max_time: Option<Duration>,
 
+    /// The server normally times out idle cursors after an inactivity period to prevent excess
```
[the remark this answers is not in the document]
Comment (isabelatkinson) [reply src/gridfs/options.rs]: Ah I had forgotten about that, I thought we had missed it in the first pass. Removed.

File: src/gridfs.rs (hunk 1 above)
Comment (isabelatkinson) [reply src/gridfs.rs:337]: yep, just verified in the shell

**Review by kmahar — APPROVED** [review]

**Pull request merged**